File: 03-02-2025/Typedef for a Complex Number.c
#include <stdio.h>

typedef struct {
    float real;
    float imag;
} Complex;

Complex addComplex(Complex a, Complex b) {
    Complex result;
    result.real = a.real + b.real;
    result.imag = a.imag + b.imag;
    return result;
}

int main() {
    Complex num1 = {3.5, 2.5};
    Complex num2 = {1.5, 1.5};
    Complex result = addComplex(num1, num2);
    
    printf("Result: %.2f + %.2fi\n", result.real, result.imag);
    return 0;
}

File: 03.02.2025/Typedef for a Complex Number.py
class Complex:
    def __init__(self, real, imag):
        self.real = real
        self.imag = imag

    def add(self, other):
        return Complex(self.real + other.real, self.imag + other.imag)

    def __str__(self):
        return f"{self.real} + {self.imag}i"

num1 = Complex(3.5, 2.5)
num2 = Complex(1.5, 1.5)
result = num1.add(num2)
print(f"Result: {result}")
